[Adriel kee jianwei] iP

LeirdaKJW opened the pull request:
# **NerdBot**
> 🤓 
## Have NerdBot keep track of your everyday tasks. It 
* Is Easy to use
* Is Super efficient
* Replaces your day old calendar

## All you have to do is:
1. Download it from [here](https://github.com/amoonguss1/ip/releases/tag/A-Jar)
2. Double-click the file
3. Add your tasks using the commands displayed
4. Let it manage your tasks 🤓 

And it is 🤓 **FREE**!
## Features:
- [x] Managing tasks
- [x] Managing Deadlines and Events
- [x] Reminders (coming soon)

If you are a 🤓, you can use it to practice java too. Here's the  ```main ```
```java {.line-numbers}
public static void main(String[] args) {
        new Duke("duke.txt").run();
}
```

Commit:
Add Gradle support

LeirdaKJW committed:
added greet, echo and exit

LeirdaKJW committed:
updates to greet echo, exit

LeirdaKJW committed:
implemented Add and List

LeirdaKJW committed:
added Task class and implemented marking feature

LeirdaKJW committed:
fixed a small bug caused by spaced inputs

LeirdaKJW committed:
added event, todo, deadline classes and updated the processlist method for these tasks

LeirdaKJW committed:
added test cases and fixed more white space bugs

LeirdaKJW committed:
added some exceptions and updated testcases

LeirdaKJW committed:
implemented delete feature

LeirdaKJW committed:
Add enumurations and reformat files

LeirdaKJW committed:
implement task list and add a few methods

LeirdaKJW committed:
Add a new class that abstracts the processes and reedit string processing

LeirdaKJW committed:
Add saving and loading feature and package other classes accordingly

LeirdaKJW committed:
reformat and remove redundant lines

LeirdaKJW committed:
package relevant classes

LeirdaKJW committed:
Add command date and process dates properly

LeirdaKJW committed:
Fix a small bug in Event

LeirdaKJW committed:
merge save and load feature

LeirdaKJW committed:
Merge branch 'branch-Level-8'

LeirdaKJW committed:
Revert "Fix a small bug in Event"

This reverts commit c1550812133f86d2cfd906373db75cfe7476a3c3.

LeirdaKJW committed:
Add save and load feature

LeirdaKJW committed:
implement date and time functions for event and deadline

LeirdaKJW committed:
Merge branch 'branch-Level-7'

LeirdaKJW committed:
Merge branch 'branch-Level-8'

LeirdaKJW committed:
Change processCommands to UI

LeirdaKJW committed:
Add more oop and reorganised files

LeirdaKJW committed:
Fix loading and saving bugs and refactor files into packages

LeirdaKJW committed:
Merge branch 'add-gradle-support'

LeirdaKJW committed:
Add gradle support for testing and building

LeirdaKJW committed:
Add junit tests for todo and Event classes

LeirdaKJW committed:
Fix invalid input exceptions

LeirdaKJW committed:
Add jdocs to everything

LeirdaKJW committed:
Merge branch 'branch-A-JavaDoc'

LeirdaKJW committed:
Add available commands display and tag jdocs

LeirdaKJW committed:
rename variables and methods to follow coding standards

LeirdaKJW committed:
Merge branch 'branch-A-CodingStandard'

LeirdaKJW committed:
Add find command and fix empty description bug

LeirdaKJW committed:
Fix some exception bugs

LeirdaKJW committed:
change data file

LeirdaKJW committed:
Merge branch 'branch-Level-9'

Yufannnn commented on src/main/java/Duke/Duke.java:
```diff
@@ -0,0 +1,103 @@
+package Duke;
```
I like how your code is organized into different modules for easy maintenance. The use of polymorphism for different commands is also a smart implementation. :wink:

Yufannnn commented on src/main/java/Duke/Parser/Parser.java:
```diff
@@ -0,0 +1,194 @@
+     * @param input The full line of the command and arguments.
+     * @return The date of the deadline.
+     */
+    public String parseDeadline(String input) {
```
Any reason for these parse methods only returns Strings instead of directly returning the Todo, Deadline, or Event that was created during parsing? :grinning:

Yufannnn commented on src/main/java/Duke/Parser/Parser.java:
```diff
@@ -0,0 +1,194 @@
+package Duke.Parser;
```
I like how your code uses a switch statement to handle different cases of commands and using an enum to map strings to corresponding command types. :thumbsup:

Yufannnn commented on src/main/java/Duke/entities/Deadline.java:
```diff
@@ -0,0 +1,49 @@
+     * @return a String representing the Deadline to be saved.
+     */
+    @Override
+    public String toSave() {
```
I like how you use String.format() to make the toString method concise and clean.

Yufannnn commented on src/main/java/Duke/Commands/Command.java:
```diff
@@ -0,0 +1,15 @@
+    /**
+     * Duke.Commands abstract method that processes the command.
+     */
+    public abstract void processCommand();
```
Any reason why not directly declare a public abstract void process command (TaskList list, String desc, String by, Ui ui) in the Command class, with signature: TaskList list, String desc, String by, Ui ui?

Yufannnn commented on src/main/java/Duke/Parser/Parser.java:
```diff
@@ -0,0 +1,194 @@
+package Duke.Parser;
+
+import Duke.Commands.*;
```
Should we try to avoid using wildcards in the import statement?

Yufannnn commented on src/main/java/Duke/enums/CommandEnums.java:
```diff
@@ -0,0 +1,33 @@
+package Duke.enums;
```
I like how you use an enum here to ensure type safety.

Yufannnn reviewed (commented):
Although the complexity of the code presents a challenge, it is evident that the structure is well thought-out and the readability of the code is generally good. However, upon closer examination, some sections of the code seem overly complex and could benefit from simplification, which would enhance its maintainability and overall quality.

panpannnnn commented on src/main/java/Duke/Commands/DeadlineCommand.java:
```diff
@@ -0,0 +1,32 @@
+ */
+public class DeadlineCommand extends Command {
+    @Override
+    public void processCommand() {
```
Is this function necessary? what is the use of it?

panpannnnn commented on src/main/java/Duke/Commands/EventCommand.java:
```diff
@@ -0,0 +1,33 @@
+    public void processCommand(TaskList list, String desc, String from, String to, Ui ui) {
+        Event event = new Event(desc, from, to);
+        list.addTask(event);
+        ui.print(String.format("Sure!, I've added the following events:\n %s", event.toString()));
```
maybe can add more customisations so that it would look nicer

panpannnnn commented on src/main/java/Duke/Commands/FindCommand.java:
```diff
@@ -0,0 +1,38 @@
+        ui.print("Here are tasks that associate with " + item + ":");
+
+        for (int i = 0; i < list.getSize(); i++) {
+            boolean toPrint = false;
```
Might be easier to come up with a system that doesn't require so much processing.

panpannnnn commented on src/main/java/Duke/Duke.java:
```diff
@@ -0,0 +1,103 @@
+ * Running a duke object loads data from the specified file into memory,
+ * and exiting the program writes data to the hard disk.
+ */
+public class Duke {
```
Very neat and easy to read and understand

panpannnnn commented on src/main/java/Duke/Duke.java:
```diff
@@ -0,0 +1,103 @@
+        while (input.equals("bye") == false) {
+            try {
+                Command command = parser.parseCommand(input);
+                if (command instanceof ListCommand) {
```
Maybe can use case and switch instead for it to look neater

panpannnnn commented on src/main/java/Duke/Ui/Ui.java:
```diff
@@ -0,0 +1,71 @@
+        printDivider();
+    }
+
+    public void showCommandList() {
```
Nice addition for better user experience.

panpannnnn commented on src/main/java/Duke/entities/Todo.java:
```diff
@@ -0,0 +1,39 @@
+package Duke.entities;
```
Also I like how you packaged the different entities, it makes your repo look more organised.

panpannnnn reviewed (commented):
Very clean and organised code:)

LeirdaKJW committed:
Add gui package and update overloading process

LeirdaKJW committed:
Add a simple gui to the nerdbot

LeirdaKJW committed:
Refactor Commands and nerd to suit the gui.

LeirdaKJW committed:
Add landing message and auto exit app

LeirdaKJW committed:
Add assertions for task and tasklist

LeirdaKJW committed:
Merge pull request #1 from amoonguss1/branch-A-Assertions

Add assertions for task and tasklist

LeirdaKJW committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/Duke.java
#	src/main/java/entities/Deadline.java
#	src/main/java/entities/Event.java
#	src/main/java/entities/Task.java
#	src/main/java/entities/Todo.java

LeirdaKJW committed:
Revert "Merge branch 'branch-Level-7'"

This reverts commit 6427dc7e3d65a15292a4eb54b9ec9dbb8ba51050.

LeirdaKJW committed:
Improve code quality and SlAP the commands.

There are several process command methods in the command classes that did
not apply SLAP as well as possible.

By abstracting out more complexity from the methods, other users will be able to read it with lesser ambiguity.

Complexity is handled by the ui and parser classes and the command classes will be a level of abstraction higher.

LeirdaKJW committed:
Merge pull request #2 from amoonguss1/A-CodeQuality

A-Code-Quality

LeirdaKJW committed:
Add reminder commands to the Nerdbot.

Nerdbot currently cannot send reminders to the user on upcoming deadlines and events.

This feature will add reminders for the user on upcoming deadlines based on the range of days from the current day.

By adding on to the existing command list, we can consistently update Nerdbot with new features easily, without changing any main structure of the program.

LeirdaKJW committed:
Merge pull request #3 from amoonguss1/branch-B-Reminders

Add reminder commands to the Nerdbot.

LeirdaKJW committed:
anchor text box to the mainwindow and restyle boxes

LeirdaKJW committed:
Merge branch 'branch-A-BetterGui'

LeirdaKJW committed:
Add connection message and rename classes

LeirdaKJW committed:
add wallpaper and icons to the gui

LeirdaKJW committed:
Merge branch 'branch-A-Personality'

LeirdaKJW committed:
Reorganise the Exception handling in parseCommands

LeirdaKJW committed:
Refactor classes and jdocs

LeirdaKJW committed:
Add junit test for deadline

LeirdaKJW committed:
Add find command in default message

LeirdaKJW committed:
Update README.md

LeirdaKJW committed:
Update README.md

LeirdaKJW committed:
Update README.md

LeirdaKJW committed:
Update README.md

LeirdaKJW committed:
Update Ui.java

LeirdaKJW committed:
Update README.md

LeirdaKJW committed:
Fix date searching

LeirdaKJW committed:
Merge branch 'master' of https://github.com/amoonguss1/ip

LeirdaKJW committed:
Update README.md

LeirdaKJW committed:
Update README.md

LeirdaKJW committed:
Add files via upload

LeirdaKJW committed:
Refactor image

LeirdaKJW committed:
Change GUI colors and update

LeirdaKJW committed:
Fix storage bug when loading a new Nerdbot

LeirdaKJW committed:
Add message for empty lists

LeirdaKJW committed:
Fix the exit bug
